t/lib/proxy-wasm-tests/hostcalls/src/filter.rs

```diff
@@ -139,6 +139,19 @@ impl HttpContext for TestHttp {
 
     fn on_http_response_body(&mut self, len: usize, eof: bool) -> Action {
         info!("[hostcalls] on_response_body, {} bytes, eof: {}", len, eof);
+
+        if let Some(max) = self.body_max {
+            self.body_seen += len;
+            if self.body_seen > max {
+                self.set_http_response_body(max, 0, "".as_bytes());
+                if max == 0 {
+                    return Action::Pause;
```

Comment:
Ok I see now; if this is currently working then it is most likely per accident, I don't think the test cases are properly set up to catch what may be going on here. What this is doing is enabling the response buffering feature (returning `PAUSE` from `response_body`), so this feels like something is off here.

Reply:
Thanks for the review!! I'll look at the V8 CentOS issue today then get back here to apply your suggested changes.

Reply:
> I've had success so far with the approach I wrote in the PR review

Just to clarify: are you going to carry on implementing it or do you want me to try my hand at it? If you have a branch with those changes, even if in wip state, I can try running it on the Gateway with my modified version of the Coraza filter to check how that is behaving (or if the filter needs further changes).

Reply:
I am still spending some time trying to make the body work since yesterday although I got sick this week, slow progress; but I can push it later.

Reply:
Yesterday after the call it kind of went downhill and I could not even sit at the computer anymore. So, maybe today or maybe early next week now.

Reply:
Ok I just succeeded replacing the body as well with the same approach. Lots of polishing + a few more tests needed now.